core-bundle/src/Resources/contao/classes/Versions.php

```diff
@@ -759,32 +759,36 @@ protected function getEditUrl()
 			return sprintf($this->strEditUrl, $this->intPid);
 		}
 
-		$strUrl = Environment::get('request');
+		$pairs = array();
+		$request = System::getContainer()->get('request_stack')->getCurrentRequest();
 
-		// Save the real edit URL if the visibility is toggled via Ajax
-		if (preg_match('/&(amp;)?state=/', $strUrl))
+		parse_str($request->server->get('QUERY_STRING'), $pairs);
```

Comment:
Is there a specific reason we don’t use `$request->query->all()` here?

Reply:
Good point. We are doing it this way here:

https://github.com/contao/contao/blob/ab71fd99f6c7ce36595a0d8dd75b17710d98a2bb/core-bundle/src/Resources/contao/library/Contao/Controller.php#L1119

https://github.com/contao/contao/blob/ab71fd99f6c7ce36595a0d8dd75b17710d98a2bb/core-bundle/src/Resources/contao/modules/ModuleLogout.php#L73

https://github.com/contao/contao/blob/ab71fd99f6c7ce36595a0d8dd75b17710d98a2bb/core-bundle/src/Resources/contao/pages/PageLogout.php#L62

But maybe we should use `$request->query->all()` everywhere instead?

Reply:
But not in a bugfix version then I think. I’d merge this as is and maybe change it in Contao 5.

Reply:
Yeah that looks really weird but I agree changing it in 5.